nrf-hal-common/src/uarte.rs

```diff
@@ -676,7 +672,7 @@ where
             }
             Ok(self.rx_buf[0])
         } else {
-            start_read(&uarte, self.rx_buf, 1)?;
+            start_read(&uarte, &mut self.rx_buf[..1])?;
```

Comment:
If we only ever use the first byte of the buffer, I don't see why we shouldn't take a `&'static mut [u8; 1]` in `split` instead of a slice

Reply:
Yeah, we could. Though if we don't, we can potentially extend the UarteRx API later with a read() taking a rx_buffer and a timer without having to modify split(). I'm fine either way.

Reply:
@jonas-schievink I've updated it to require a slice of size 1, I guess changing the API of split() to take any slice at a later point will not break anything.